sdk/resources/Azure.ResourceManager.Resources/src/Generated/ArmDeploymentStackData.cs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Resources.Models;
+
+namespace Azure.ResourceManager.Resources
+{
+    /// <summary>
+    /// A class representing the ArmDeploymentStack data model.
+    /// Deployment stack object.
+    /// </summary>
+    public partial class ArmDeploymentStackData : TrackedResourceData
```

Comment:
The generator determines it is TrackResource or regular Resource (ProxyResource included) by the properties defined on its model.
If the resource model has `id` `type` `name` `location` and `tags`, the generator will always generate the resource model as `TrackedResourceData`.
We might solve this issue when it has all 5 properties but in fact it is not tracked resource when we could convert your swagger into tsp and generate from tsp as input.
But now, if you want this to be inherited from `ResourceData`, please add a partial class of this class in the custom directory (or customization or whatever similar and existed) like this:
```csharp
namespace Azure.ResourceManager.Resources
{
    public partial class ArmDeploymentStackData : ResourceData
    {}
}
```
You do not have to put anything inside it. After you have this partial class (ignore the compilation error), regenerate the library, and the generator will pick up your new base type in the generated code.

Please make sure that you make the change because of the fact that this resource is not tracked resource, not from alignment. Because ArmDeploymentData inherits from `ResourceData` might be historical, for example, this might be caused by mistake, and when we fix the mistake, the library has been released and unfortunately we cannot change that any more. This does not mean we have to do the same here to align with a mistake.

Reply:
Fixed